x-pack/plugins/apm/server/lib/settings/custom_link/custom_link_types.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { CustomLinkES } from './create_custom_link_index';
+
+interface Filter {
+  key: string;
+  value: string[];
+}
+
+export interface CustomLink {
+  id?: string;
+  '@timestamp'?: number;
+  label: string;
+  url: string;
+  filters: Filter[];
+}
+
+export function convertTo(customLinkES: CustomLinkES[]): CustomLink[] {
```

Comment:
I think this function should be co-located with the other `convertTo` function. And they should probably be called something like `convertTo` / `convertFrom`.  Or `toESFormat` / `fromESFormat`. Or whatever you think makes sense